### PR TITLE
Add Edges table to resource detail view of dashboard

### LINKED ISSUE #2709: Expose identity of edges in the dashboard
### What problem are you trying to solve?

See #2708

### How should the problem be solved?

For dashboard users, exposing identity of edges in an "Identity" table. In the future, there could be a dedicated "Identity" view page, but initially this feature will be only a table.

### Any alternatives you've considered?

See #2708

### How would users interact with this feature?

An `Identity` table will appear in the `Overview` view of the dashboard for each namespace, and in every resource detail view. 

The table will contain these columns (as depicted above and in the CLI as `linkerd edges identity`):
    - SRC
    - DST
    - Client
    - Server
    - Message/Reason (We will experiment with the best way to show this for readability/usefulness. Exposing the entire reason may be hard to parse, but obscuring it in favor of a tooltip creates more work for the user.)

grambp

This issue builds off ideas in #2605 and #2606.

### PR DESCRIPTION
**Pull request by scottcarol**
This PR adds an Edges table to the resource detail view if one or more edges are returned going to/from that specific resource. A tooltip icon to the right of the table title explains what "Edges" data is. If a specific resource is not part of any edges, the Edges table should not show.

You can check the data by running `linkerd edges`.

Screen Shot 2019-06-18 at 2 27 32 PM

Resolves #2709.

### CODE REVIEW

**Comment by grampelberg** [comment]
@scottcarol can the source/dst names be links? client/server would be nice too, but I don't think we have a way to display service accounts yet.

**Comment by scottcarol** [comment]
@grampelberg yup, can do.

File: web/app/js/components/ResourceDetail.jsx [hunk 1]
```diff
@@ -14,15 +15,19 @@ import TopRoutesTabs from './TopRoutesTabs.jsx';
...
 import _isNil from 'lodash/isNil';
 import _merge from 'lodash/merge';
 import _reduce from 'lodash/reduce';
+import { processEdges } from './util/EdgesUtils.jsx';
 import { withContext } from './util/AppContext.jsx';
 
 // if there has been no traffic for some time, show a warning
 const showNoTrafficMsgDelayMs = 6000;
+// resource types supported when querying API for edge data
+const possibleResourceType = ["daemonset", "deployment", "job", "pod", "replicationcontroller", "statefulset"];
```
Comment (rmars) [inline web/app/js/components/ResourceDetail.jsx]: nit: can we call this "edgeDataAvailableType" or `edgeDataAvailable` or something? `possible` seems kind of vague

File: web/app/js/components/ResourceDetail.jsx [hunk 2]
```diff
@@ -347,6 +365,18 @@ export class ResourceDetailBase extends React.Component {
           title="TCP"
           isTcpTable={true}
           metrics={this.state.podMetrics} />
+
+        {
+          noEdges ? null :
```
Comment (rmars) [inline web/app/js/components/ResourceDetail.jsx]: if you're not using `noEdges` anywhere else, might as well just directly say `_isEmpty(this.state.edges)` here

File: web/app/js/components/EdgesTable.jsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      dataIndex: "message",
+      isNumeric: false,
+      filter: d => d.noIdentityMsg,
+      render: d => d.noIdentityMsg,
+      sorter: d => d.noIdentityMsg
+    }
+  ];
+};
+
+const tooltipText = "Edges show the source, destination name and identity " +
```
Comment (dadjeibaah) [inline web/app/js/components/EdgesTable.jsx]: Super minor nit: You could do use backticks for this string. This will allow you to split the line if its too long without having to string concatenate.
```diff
diff --git a/web/app/js/components/EdgesTable.jsx b/web/app/js/components/EdgesTable.jsx
index ebf46541..48bdfb58 100644
--- a/web/app/js/components/EdgesTable.jsx
+++ b/web/app/js/components/EdgesTable.jsx
@@ -71,8 +71,8 @@ const edgesColumnDefinitions = (PrefixedLink, namespace, type) => {
   ];
 };
 
-const tooltipText = "Edges show the source, destination name and identity " +
-  "for proxied connections. If no identity is known, a message is displayed.";
+const tooltipText = `Edges show the source, destination name and identity 
+  for proxied connections. If no identity is known, a message is displayed.`;
 
 class EdgesTable extends React.Component {
   static propTypes = {
```
